extensions/jdbc/jdbc-postgresql/deployment/src/main/java/io/quarkus/jdbc/postgresql/deployment/JDBCPostgreSQLProcessor.java

```diff
@@ -42,11 +43,14 @@ void configureAgroalConnection(BuildProducer<AdditionalBeanBuildItem> additional
     }
 
     @BuildStep
-    void registerServiceBinding(Capabilities capabilities, BuildProducer<ServiceProviderBuildItem> serviceProvider) {
+    void registerServiceBinding(Capabilities capabilities,
+            BuildProducer<ServiceProviderBuildItem> serviceProvider,
+            BuildProducer<DefaultDataSourceDbKindBuildItem> dbKind) {
         if (capabilities.isPresent(Capability.KUBERNETES_SERVICE_BINDING)) {
             serviceProvider.produce(
                     new ServiceProviderBuildItem("io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConverter",
                             PostgreSqlServiceBindingConverter.class.getName()));
+            dbKind.produce(new DefaultDataSourceDbKindBuildItem(DatabaseKind.POSTGRESQL));
```

Comment:
Hmmm? This is a bit weird. How will it work with the other JDBC drivers if you have several around?

Typically what if you have multiple datasources targeting different RDBMS?

Reply:
For the time being, our service binding support only works with the default datasource.

We probably could make it work for other datasource as well, but the problem is that these are not known at build time, so it's a much better experience for the typical use case to just configure a default datasource

Reply:
I really think we should make the simple case of having a single datasource easy to use with the service binding and get some real world experience before we opt to design a more comprehensive solution

Reply:
OK but then, having an `Optional` really wouldn't work.

I think you need a `List`, you need to produce one build item in each driver and then only push the value if there's only one element in the list.

Reply:
Sure, that makes sense

Reply:
PR updated

Reply:
WDYT about it now?

If you think it's OK, I can add an integration test and take it out of draft

Reply:
The logic looks good now. Just needs to be pushed to the other drivers and we should be good to go.

Reply:
Will do. 

I'll also add an integration test for the k8s service binding because we currently don't have one

Reply:
Done